File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp1, dp2 = 0 , cost[0]

        for i in range(1 , len(cost)):
            temp = dp1
            dp1 = dp2
            dp2 = cost[i] + min(temp, dp1)

        return min(dp1,dp2)

File: Data Structures & Algorithms/climbing-stairs/submission-1.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp1 = 1, dp2 = 2;
        if (n == 1) return 1;
        if (n == 2) return 2;
        for (int i = 3; i < n; i++) {
            int temp = dp1;
            dp1 = dp2;
            dp2 = temp + dp2;
        }
        return dp1 + dp2;
    }
};
